### PR TITLE
Add optimization capability for OMPL freespace planner

### PR DESCRIPTION
**Pull request by Levi-Armstrong**
Before we were using OMPL OptimizePlan but there is an issue with hybridization. This add she same functionality but with hybridization disabled. 

Update:
This PR also includes restructuring of the OMPL planner interfaces to mimic the TrajOpt Planner interface. Now OMPL interface is a single planner with different configurations for freespace. This now will allow for OMPL to be reconfigured without having to create a new motion planner. @mpowelson

### CODE REVIEW

File: tesseract/tesseract_planning/tesseract_motion_planners/include/tesseract_motion_planners/ompl/impl/ompl_freespace_planner.hpp [hunk 1]
```diff
@@ -69,11 +68,54 @@ tesseract_common::StatusCode OMPLFreespacePlanner<PlannerType>::solve(PlannerRes
...
-  // Solve problem. Results are stored in the response
-  // Disabling hybridization because there is a bug which will return a trajector that starts at the end state
-  // and finishes at the end state.
-  ompl::base::PlannerStatus status =
-      parallel_plan_->solve(config_->planning_time, 1, static_cast<unsigned>(config_->max_solutions), false);
+  ompl::base::PlannerStatus status;
+  if (!config_->optimize)
+  {
+    // Solve problem. Results are stored in the response
+    // Disabling hybridization because there is a bug which will return a trajector that starts at the end state
```
Comment (mpowelson) [inline tesseract/tesseract_planning/tesseract_motion_planners/include/tesseract_motion_planners/ompl/impl/ompl_freespace_planner.hpp]: spelling `trajector` -> `trajectory`

**Comment by Levi-Armstrong** [comment]
This change will make it easy to create a constrained ompl planner leveraging TrajOpt to project the sample state onto the constraint manifold.

File: tesseract/tesseract_planning/tesseract_motion_planners/include/tesseract_motion_planners/ompl/config/ompl_planner_config.h [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+   * This is required because not all OMPL planners are optimize graph planners. If the planner you choose is an
+   * optimize graph planner then setting this to true has no affect. In the case of non-optimize planners they still
+   * use the OptimizeObjective function but only when searching the graph to find the most optimize solution based
+   * on the provided optimize objective function. In the case of these type of planners like RRT and RRTConnect if set
+   * to true it will leverage all planning time to keep finding solutions up to your max solutions count to find the
+   * most optimal solution.
+   */
+  bool optimize = false;
+
+  /** @brief Trajopt problem to be solved ***REQUIRED*** */
```
[the remark this answers is not in the document]
Comment (Levi-Armstrong) [reply tesseract/tesseract_planning/tesseract_motion_planners/include/tesseract_motion_planners/ompl/config/ompl_planner_config.h]: Fixed

File: tesseract/tesseract_planning/tesseract_motion_planners/include/tesseract_motion_planners/ompl/config/ompl_planner_freespace_config.h [hunk 3]
```diff
@@ -0,0 +1,101 @@
+/**
+ * @file ompl_planner_config.h
```
[the remark this answers is not in the document]
Comment (Levi-Armstrong) [reply tesseract/tesseract_planning/tesseract_motion_planners/include/tesseract_motion_planners/ompl/config/ompl_planner_freespace_config.h]: Fixed

File: tesseract/tesseract_planning/tesseract_motion_planners/include/tesseract_motion_planners/ompl/impl/ompl_freespace_planner.hpp (hunk 1 above)
Comment (Levi-Armstrong) [reply tesseract/tesseract_planning/tesseract_motion_planners/include/tesseract_motion_planners/ompl/impl/ompl_freespace_planner.hpp]: fixed

**Pull request merged**